arch/arm/crc32_acle.c

```diff
@@ -5,39 +5,88 @@
  *
 */
 
-#ifdef __ARM_FEATURE_CRC32
-#include <arm_acle.h>
-#ifdef ZLIB_COMPAT
+#if defined(__ARM_FEATURE_CRC32)
+# include <arm_acle.h>
+# ifdef ZLIB_COMPAT
 #  include <zconf.h>
-#else
+# else
 #  include <zconf-ng.h>
-#endif
-#ifdef __linux__
+# endif
+# ifdef __linux__
 #  include <stddef.h>
-#endif
+# endif
+#include "zutil.h"
 
 uint32_t crc32_acle(uint32_t crc, const unsigned char *buf, uint64_t len) {
-    register uint32_t c;
-    register const uint16_t *buf2;
-    register const uint32_t *buf4;
+    uint32_t c = ~crc;
+    const uint16_t *buf2;
+    const uint32_t *buf4;
+    const size_t sz2 = sizeof(uint16_t);
+    const size_t sz4 = sizeof(uint32_t);
 
     c = ~crc;
     if (len && ((ptrdiff_t)buf & 1)) {
         c = __crc32b(c, *buf++);
         len--;
     }
 
-    if ((len > 2) && ((ptrdiff_t)buf & 2)) {
-        buf2 = (const uint16_t *) buf;
-        c = __crc32h(c, *buf2++);
+    if ((len > sz2) && ((ptrdiff_t)buf & sz2)) {
+        uint16_t val;
+        MEMCPY(&val, buf, sizeof(val));
```

Comment:
I still don't see any valid reason to use MEMCPY() as buf is guaranteed to be always aligned.

Reply:
Why are you using sz2 and sz4 variables? Is there some performance improvement over just using sizeof() in all those places?

Reply:
I fixed both issues.  Thanks.